From a019b96eac259ad13c9f4a81f4a130c3b83b0db5 Mon Sep 17 00:00:00 2001
From: Niall Creech <niallcreech@googlemail.com>
Date: Tue, 4 May 2010 05:21:37 +0200
Subject: [PATCH] Set info file magic and type TTFNheader fields in two steps

The current code is perfectly legal, even though a bit ugly, but this
fixes an abort when using the fortify compiler support, as it exceeds
its bound checks.
---
 src/xfstt.cc |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/xfstt.cc b/src/xfstt.cc
index d02f2d9..4eddd02 100644
--- a/src/xfstt.cc
+++ b/src/xfstt.cc
@@ -313,9 +313,10 @@ ttSyncAll(int gslist = 0)
 	}
 
 	TTFNheader info;
-	strncpy(info.magic, "TTFNINFO", 8);
+	strncpy(info.magic, "TTFN", 4);
 	info.version = TTFN_VERSION;
 	info.crc = 0;	// XXX
+	strncpy(info.type, "INFO", 4);
 	fwrite((void *)&info, 1, sizeof(info), infoFile);
 	strncpy(info.type, "NAME", 4);
 	fwrite((void *)&info, 1, sizeof(info), nameFile);
-- 
1.7.1

